evm/utils/address.py

```diff
@@ -16,3 +20,15 @@ def force_bytes_to_address(value):
 
 def generate_contract_address(address, nonce):
     return keccak(rlp.encode([address, nonce]))[-20:]
+
+
+def generate_create2_contract_address(salt, code):
+    """
+    If contract is created by transaction, `salt` should be empty.
```

Comment:
Shouldn't in this case the salt be specified by the transaction somehow (maybe the first 32 bytes of `data`)? Otherwise one can't deploy the same contract twice, or am I missing something?

Reply:
The `salt` could be transaction nonce if `nonce` is not abstracted away eventually. Or if it is, one could simply append data to `code` and treat it as `salt`.